fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -131,27 +132,27 @@
     // The functionality is split into a new class.
     private static final Sort sort = new Sort();
 
+    private static CompartmentUtil compartmentHelper;
```

Comment:
I added this note at the class level:
> Initialization is *not* thread safe, so care must be taken to avoid calling initialization while other threads are using the class.

Not very satisfying though.  Can you think of a better way without incurring the pain of changing every single spot that is using SearchUtil statically?

Reply:
Yeah you can do the inner static class pattern to make the singleton initialization thread-safe...
```
    private static class Holder {
        private static final CompartmentUtil INSTANCE = new CompartmentUtil();
    }

   private CompartmentUtil getCompartmentHelper() {
       return Holder.INSTANCE;
```

Reply:
we needed a way to re-initialize the CompartmentUtil for each different module on the system.  the only way i could find to do that in a thread-safe way was to "bite the bullet" and refactor SearchUtil...it is no longer static (nor a singleton instance) and so that should address the concern here.  however, this forced numerous other changes/decisions and so a new review is needed